store/driver/tikv_driver.go

```diff
@@ -331,3 +334,20 @@ func (s *tikvStore) CurrentVersion(txnScope string) (kv.Version, error) {
 func (s *tikvStore) ShowStatus(ctx context.Context, key string) (interface{}, error) {
 	return nil, kv.ErrNotImplemented
 }
+
+// GetLockWaits get return lock waits info
+func (s *tikvStore) GetLockWaits() ([]*deadlockPB.WaitForEntry, error) {
+	stores := s.GetRegionCache().GetStoresByType(tikvrpc.TiKV)
+	var resp *tikvrpc.Response
+	var err error
+	for _, store := range stores {
+		resp, err = s.GetTiKVClient().SendRequest(context.TODO(), store.Addr, tikvrpc.NewRequest(tikvrpc.CmdLockWaitInfo, &kvrpcpb.GetLockWaitInfoRequest{}), time.Second*30)
+		if err == nil {
+			break
+		}
+	}
+	if resp == nil {
+		return nil, err
+	}
+	return resp.Resp.(*kvrpcpb.GetLockWaitInfoResponse).Entries, nil
+}
```

Comment:
I'm just noticing that there are several problems:
* If the cluster is large and the transactions conflict very frequently, is it possible that the amount of lock waiting will be too large, and consumes too much memory in TiDB? Perhaps the problem is not that bad since the deadlock detector works well.
* We didn't set a limit of entries returned by the GetLockWaitInfo request. But is it possible that there are too many lock wait entries in a single TiKV node, and scanning so many entries occupies the lock manager's thread for too long and increases the node's latency?

PTAL @youjiali1995

Reply:
I think it's ok for now, but we should file issues for potential risks of the feature.